rust/cubesql/cubesql/src/compile/engine/df/scan.rs

```diff
@@ -1135,10 +1135,12 @@ pub fn transform_response<V: ValueObject>(
                                     ))
                                 })?;
                             // TODO switch parsing to microseconds
-                            if timestamp.timestamp_millis() > (((1 as i64) << 62) / 1_000_000) {
+                            if timestamp.timestamp_millis() > (((1i64) << 62) / 1_000_000) {
                                 builder.append_null()?;
+                            } else if let Some(nanos) = timestamp.timestamp_nanos_opt() {
+                                builder.append_value(nanos)?;
                             } else {
-                                builder.append_value(timestamp.timestamp_nanos_opt().unwrap())?;
+                                builder.append_null()?;
```

Comment:
Another problem is keeping logs clean. We need to log it. Without that, it will be a silent surprise that some values in the column are null.

Reply:
Yeah, agree! Thnx for pointing! Added err logging!